### ISSUE TITLE
Mopidy v1.0 service doesn't start when log directory doesn't exist

### ISSUE THREAD

**Issue by gvc0** [mopidy/mopidy#1113]
I just updated to version 1.0 and the Mopidy daemon didn't start anymore after reboot. Root cause is that I located '/var/log' in RAM (tmpfs). As a result the subdirectory 'mopidy' doesn't exist anymore after reboot. I didn't have this issue with previous versions of Mopidy. (I'm using Ubuntu 12.04).

Currently I made a workaround by creating the directory and configuring permissions in the daemon start-up script '/etc/init.d/mopidy'. It would be nice to have a log file path setting in '/etc/mopidy/mopidy.conf'.

**Comment by adamcik** [comment]
@jodal I'll leave it to you if we want to do something with this in mopidy, though unlikely given the logging config. So probably this is something to fix just in the deb.

**Comment by jodal** [comment]
The Debian package creates `/var/log/mopidy` when the package is installed. This is not new in 1.0.

Mopidy can't create this directory itself, as it runs as the user `mopidy` and it doesn't have write access to `/var/log`.

If the Debian package was to fix this, it would need to create the dir and give it the right permissions as part of the startup script, where one have root access. This is of course easy with sysvinit scripts, and possible with systemd, but it feels like a hack.

How does other services that run as non-root handle creation of dirs in `/var/log` when it is a tmpfs?

Long term (as when most systems are running systemd) I guess the solution is simply to just log to stdout/journalctl and not to file at all.

**Comment by jodal** [comment]
@gvc0 I have a guess at why you see this issue with 1.0: You've previously commented out (or removed, or whatever) the logging section in `/etc/mopidy/mopidy.conf`. That way you turned off logging to `/var/log/mopidy/mopidy.log`. With 1.0, some other default config was changed, and you got a question about merging your changes and the package's new default config file when you upgraded. You probably let it overwrite the config with the default. You can find the old config file as `/etc/mopidy/mopidy.conf.dpkg-old`.

**Comment by gvc0** [comment]
I took a backup of my mopidy.conf before upgrading. After some testing it turns out the logging setting config_file is of importance. It was commented out before, which actualy disables logging. After the upgrade it was set to config_file = /etc/mopidy/logging.conf. I managed to change the log location in this config file by changing the setting args = ('/var/log/mopidy.log',). Everything is working fine now (and Mopidy creates and logs to the new log file).

Maybe some of this is worth mentioning in the documentation. Thanks for your help.
